src/filesystem.cc

```diff
@@ -948,28 +969,29 @@ ASFileSystem::ParsePath(
   return Status::Success;
 }
 
-ASFileSystem::ASFileSystem(const std::string& path) : as_regex_(AS_URL_PATTERN)
+ASFileSystem::ASFileSystem(
+    const std::string& path, const std::string& account_str,
+    const std::string& account_key)
+    : as_regex_(AS_URL_PATTERN)
 {
-  const char* account_str = std::getenv("AZURE_STORAGE_ACCOUNT");
-  const char* account_key = std::getenv("AZURE_STORAGE_KEY");
   std::shared_ptr<as::storage_account> account = nullptr;
   std::string host_name, container, blob_path, query;
   if (RE2::FullMatch(
           path, as_regex_, &host_name, &container, &blob_path, &query)) {
     size_t pos = host_name.rfind(".blob.core.windows.net");
     std::string account_name;
-    if (account_str == NULL) {
+    if (account_str.size()) {
```

Comment:
Using size() seems wrong, evaluate to "true" if the string has content
```suggestion
    if (account_str.empty()) {
```

Reply:
Yes. Updated.
The mapping from previous char NULL -> empty string, since credential cannot be empty.